### ISSUE TITLE
Feature: adding @type attribute by core

### ISSUE THREAD

**Issue by Moongazer** [sourcebroker/t3api#21]
Similar to `@id` property which is shown by default in each response, I'm asking if it make sense to have a `@type` property as well filled with the entity class name. Sure, its easy to implement on a project level, but as it seems to be a quite common property for `JSON-LD+Hydra`, I'd suggest to implement this into core. Example response:

```
{
   "@id": "/api/v1/blog/post", // optional, not really needed
   "@type": "HydraCollectionResponse", // optional, added from class-name to always have the same format
   "hydra:member": [
      {
         "name": "Blog post 1",
         "@id": "/api/v1/blog/post/1",
         "@type": "Post" // this would be useful for front-end devs (using model class-name as type)
      }, {
         "name": "Blog post 2",
         "@id": "/api/v1/blog/post/2",
         "@type": "Post"
      }
   ],
   "hydra:totalItems": 2,
   "hydra:view": {
      "hydra:first": "/api/v1/blog/post?page=1",
      "hydra:last": "/api/v1/blog/post?page=1",
      "hydra:pages": ["/api/v1/blog/post?page=1"],
      "hydra:page": 1
   },
   "hydra:search": {
      "hydra:template": "/api/v1/blog/post{?}",
      "hydra:mapping": []
   }
}
```
**Edit:** Maybe the full FQDN is better, as an entity with the same name might exist in different namespaces/contexts.

**Comment by klamparski** [comment]
@Moongazer Thanks for this input. You're absolutely right - this property should be included in core. We also already had some use cases for it. I think it will be included in next release (1.2). I also agree that FQDN is a better approach because of possible name conflicts.

**Comment by kszymukowicz** [comment]
@Moongazer 
Done in https://github.com/sourcebroker/t3api/commit/32b737260d1ac4969aa730ff98d3515a2e14d3b9

A note: `@type` is not activated by default as it can take a lot of space in the response and will be used only by little fraction of frontend applications. To activate it you need to add in your local extension `ext_localconf.php`.
```
$GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['t3api']['serializerSubscribers'][] = \SourceBroker\T3api\Serializer\Subscriber\ResourceTypeSubscriber::class;
```
